t_stress/test_stress.py

```diff
@@ -0,0 +1,427 @@
+            self.skipTest("This is an abstract class")
+        self.backends = [
+            *self.backends[:],
+            {
+                "id": "nginx-large-page",
+                "type": "nginx",
+                "port": "8000",
+                "status_uri": "http://${server_ip}:8000/nginx_status",
+                "config": NGINX_LARGE_PAGE_CONFIG,
+            },
+        ]
+        super().setUp()
+        self.create_large_page()
+
+    def tearDown(self):
+        super().tearDown()
+        self.remove_large_page()
+
+    def create_large_page(self):
+        server = self.get_server("nginx-large-page")
+        server.node.run_cmd(
+            f"fallocate -l {self.nginx_backend_page_size} {self.large_page_path}"
+        )
+
+    def remove_large_page(self):
+        server = self.get_server("nginx-large-page")
+        server.node.remove_file(str(self.large_page_path))
+
+
+class CustomMtuMixin:
+    """Mixin to set interfaces MTU values before test is started."""
+
+    def setUp(self):
+        if self._base:
+            self.skipTest("This is an abstract class")
+        super().setUp()
+        self._prev_mtu = {}
+        self.set_mtu(
+            node=remote.tempesta,
+            destination_ip=tf_cfg.cfg.get("Client", "ip"),
+            mtu=TEMPESTA_TO_CLIENT_MTU,
+        )
+        self.set_mtu(
+            node=remote.tempesta,
+            destination_ip=tf_cfg.cfg.get("Server", "ip"),
+            mtu=TEMPESTA_TO_SERVER_MTU,
+        )
+        self.set_mtu(
+            node=remote.server,
+            destination_ip=tf_cfg.cfg.get("Tempesta", "ip"),
+            mtu=SERVER_TO_TEMPESTA_MTU,
+        )
+
+    def tearDown(self):
+        # Restore previous MTU values
+        try:
+            for args in self._prev_mtu.values():
+                sysnet.change_mtu(*args)
+        finally:
+            super().tearDown()
+
+    def set_mtu(self, node, destination_ip, mtu):
+        if mtu:
+            dev = sysnet.route_dst_ip(node=node, ip=destination_ip)
+            prev = sysnet.change_mtu(node=node, dev=dev, mtu=mtu)
+            if not dev in self._prev_mtu:
+                self._prev_mtu[dev] = [node, dev, prev]
+
+
+class BaseWrkStress(
+    CustomMtuMixin, LargePageNginxBackendMixin, tester.TempestaTest, base=True
+):
+    """Base class for `wrk` stress tests."""
+
+    def start_all(self):
+        self.start_all_servers()
+        self.start_tempesta()
+        self.assertTrue(self.wait_all_connections(1))
+
+    def test_concurrent_connections(self):
+        self.start_all()
+        wrk = self.get_client("wrk")
+        wrk.set_script("foo", content='wrk.method="GET"')
+        wrk.connections = CONCURRENT_CONNECTIONS
+        wrk.duration = int(DURATION)
+        wrk.threads = THREADS
+        wrk.timeout = 0
+
+        wrk.start()
+        self.wait_while_busy(wrk)
+        wrk.stop()
+
+        self.assertGreater(wrk.statuses[200], 0)
+
+
+class WrkStress(BaseWrkStress):
```

Comment:
`WorkStress` and `TlsWrkStress` run tests double. For: `sudo ./run_tests.py t_stress.test_stress.WrkStres`
```
----------------------------------------------------------------------
Running functional tests...
----------------------------------------------------------------------

test_concurrent_connections (t_stress.test_stress.WrkStress) ... WARNING! Socket errors on wrk. Too many concurrent connections?
ok
test_concurrent_connections (t_stress.test_stress.WrkStress) ... WARNING! Socket errors on wrk. Too many concurrent connections?
ok

----------------------------------------------------------------------
```

Reply:
Fixed. It was caused by  tests_nginx.py  `WkStress` import